osf/models/preprint.py

```diff
@@ -0,0 +1,909 @@
+
+from framework.sentry import log_exception
+from osf.exceptions import (
+    PreprintStateError, InvalidTagError, TagNotFoundError
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Preprint(DirtyFieldsMixin, GuidMixin, IdentifierMixin, ReviewableMixin, UploadMixin,
+        BaseModel, Loggable, Taggable, GuardianMixin, SpamOverrideMixin, TaxonomizableMixin, ContributorMixin):
+    # Preprint fields that trigger a check to the spam filter on save
+    SPAM_CHECK_FIELDS = {
+        'title',
+        'description',
+    }
+
+    # Setting for ContributorMixin
+    DEFAULT_CONTRIBUTOR_PERMISSIONS = 'write'
+
+    provider = models.ForeignKey('osf.PreprintProvider',
+                                 on_delete=models.SET_NULL,
+                                 related_name='preprints',
+                                 null=True, blank=True, db_index=True)
+    node = models.ForeignKey('osf.AbstractNode', on_delete=models.SET_NULL,
+                             related_name='preprints',
+                             null=True, blank=True, db_index=True)
+    is_published = models.BooleanField(default=False, db_index=True)
+    date_published = NonNaiveDateTimeField(null=True, blank=True)
+    original_publication_date = NonNaiveDateTimeField(null=True, blank=True)
+    license = models.ForeignKey('osf.NodeLicenseRecord',
+                                on_delete=models.SET_NULL, null=True, blank=True)
+
+    identifiers = GenericRelation(Identifier, related_query_name='preprints')
+    preprint_doi_created = NonNaiveDateTimeField(default=None, null=True, blank=True)
+    # begin changes
+    title = models.TextField(
+        validators=[validate_title]
+    )  # this should be a charfield but data from mongo didn't fit in 255
+    description = models.TextField(blank=True, default='')
+    creator = models.ForeignKey(OSFUser,
+                                db_index=True,
+                                related_name='preprints_created',
+                                on_delete=models.SET_NULL,
+                                null=True, blank=True)
+    _contributors = models.ManyToManyField(OSFUser,
+                                           through=PreprintContributor,
+                                           related_name='preprints')
+    article_doi = models.CharField(max_length=128,
+                                            validators=[validate_doi],
+                                            null=True, blank=True)
+    files = GenericRelation('osf.OsfStorageFile', object_id_field='target_object_id', content_type_field='target_content_type')
+    primary_file = models.ForeignKey('osf.OsfStorageFile', null=True, blank=True, related_name='preprint')
+    # (for legacy preprints), pull off of node
+    is_public = models.BooleanField(default=True, db_index=True)
+    # Datetime when old node was deleted (for legacy preprints)
+    deleted = NonNaiveDateTimeField(null=True, blank=True)
```

Comment:
Could this be `deleted_on/at` to show that it's not a boolean field? and could you add an`is_deleted` property to match other models?

Reply:
@chrisseto, Steve told me to do this in one of our initial meetings.  I think we might be slowly moving to this?  The WikiPage model, for example, has a deleted field.  To check if a preprint is deleted, I see if the deleted field is populated.